### PR TITLE
gh-626: Added build checker for changelog headers.

### PR DESCRIPTION
**Pull request by MKadaner**
## Summary
What can I say. Added.

 
## References
#626

## Checklist
- [X] I have followed the [contributing guidelines](https://github.com/FarGroup/FarManager/blob/master/CONTRIBUTING.md).
- [ ] I have discussed this with project maintainers:  <br/>
If not checked, I accept that this work might be rejected in favor of a different great big ineffable plan.<br/>

### CODE REVIEW

File: misc/build-checks/hlf-checker/CommandLine.cs [hunk 1]
```diff
@@ -5,7 +5,7 @@
 using System.Linq;
 using System.Text.RegularExpressions;
 
-namespace HlfChecker
+namespace hlf_checker
```
Comment (yegor-mialyk) [inline misc/build-checks/hlf-checker/CommandLine.cs:8]: I would leave it as is, a Pascal Case is the correct (read "widely used") name convention for C# namespaces and classes.

Comment (MKadaner) [reply misc/build-checks/hlf-checker/CommandLine.cs:8]: Yeah! Just wanted to experiment with the style of project names and directories which are more in line with everything else under FarManager\misc. It looks ugly indeed. Let me change it. It's still a draft, isn't it? :)

Comment (MKadaner) [reply misc/build-checks/hlf-checker/CommandLine.cs]: So much better!

**Comment by MKadaner** [comment]
CodeFactor complains about [`Complex Method complexity = 18`](https://www.codefactor.io/repository/github/fargroup/farmanager/pull/643). I think it's a false positive. Is there any way to suppress the diagnostic in this place?

**Comment by alabuzhev** [comment]
> Is there any way to suppress the diagnostic in this place?

No idea, but the method does look somewhat complex indeed. Maybe there is a way to simplify it?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -21,20 +21,39 @@ jobs:
         working-directory: far
         run: python tools/source_validator.py
 
+#------------------------------------------------------------------------------
+  changelog-checks:
```
Comment (alabuzhev) [inline .github/workflows/build.yml:25]: Do we really need to run this and hlf-checks separately on different VMs?

Yes, running things in parallel is good in general, but in this case both checks will likely take only a few seconds, way less than the preparations.
Maybe both can be steps in one workflow?

Comment (MKadaner) [reply .github/workflows/build.yml]: Not at all. The only real advantage of having two separate checks is accounting. One can immediately see in the PR "Checks" section which check failed, `hlf` or `changelog`. It was the way to ask whether we want it for the price of two VMs. I like them combined as well.

**Comment by MKadaner** [comment]
> the method does look somewhat complex indeed.

Please see if you like it better. CodeFactor is happy, but whether the code became simpler is a matter of taste.

**Comment by alabuzhev** [comment]
Thank you.

**Pull request merged**